stl/inc/chrono

```diff
@@ -1578,16 +1578,23 @@ namespace chrono {
             "N4885 [time.hms.overview]/2 mandates Duration to be a specialization of chrono::duration.");
 
         static constexpr unsigned int fractional_width = [] {
-            auto _Num           = _Duration::period::num;
-            constexpr auto _Den = _Duration::period::den;
             // Returns the number of fractional digits of _Num / _Den in the range [0, 18].
             // If it can't be represented, 6 is returned.
             // Example: _Fractional_width(1, 8) would return 3 for 0.125.
-            _STL_ASSERT(_Num > 0 && _Den > 0, "Numerator and denominator can't be less than 1.");
-            unsigned int _Result = 0;
-            for (; _Num % _Den != 0 && _Result < 19; _Num = _Num % _Den * 10, ++_Result) {
+            auto _Den = _Duration::period::den;
+            _STL_ASSERT(_Duration::period::num > 0 && _Den > 0, "Numerator and denominator can't be less than 1.");
+            unsigned int _Power_of_2_in_den = 0u;
+            unsigned int _Power_of_5_in_den = 0u;
+            for (; _Den % 2 == 0; _Den /= 2, ++_Power_of_2_in_den) {

```

Comment:
We can make this faster by doing a `countr_zero`. (although I'm not that concerned about it)

Reply:
Good observation. I believe there's nothing to do here because we're immediately computing a `static constexpr` so everything happens at compile-time, and there is no significant throughput gain to be had here.